client/state/sites/vouchers/actions.js

```diff
@@ -103,3 +113,33 @@ export function requestSiteVouchers( siteId ) {
 			} );
 	};
 }
+
+/**
+ * Assign a voucher to the given site.
+ *
+ * @param {Number} siteId - identifier of the site
+ * @param {String} serviceType - service type
+ * @returns {Function} a promise that will resolve once fetching is completed
+ */
+export function assignSiteVoucher( siteId, serviceType ) {
+	return dispatch => {
+		dispatch( vouchersRequestAction( siteId ) );
+
+		return wpcom
+			.site( siteId )
+			.adCreditVouchers()
+			.assign( serviceType )
+			.then( data => {
+				const { voucher = {} } = data;
```

Comment:
Fun fact!

If all you are doing with this is destructuring it, you can do that in the function definition…

```js
.then( ( { voucher = {} } ) => {
	dispatch( … );
	dispatch( … );
} );
```